### PR TITLE
Reorg of flow tests. Moving from rmutil to RLTest. Enable replication and aof test in CI.

### LINKED ISSUE #595: tests/test_module.py failing to compile due to Python version 3 not supporting trailing 'L
Python version 3.8 does not support a trailing 'L' 
There are hundreds of lines with trailing L and Python version 3.0 does not support this syntax anymore on /RedisTimeSeries/tests/test_module.py

See https://docs.python.org/3/whatsnew/3.0.html#integers for more details

_The repr() of a long integer doesn’t include the trailing L anymore, so code that unconditionally strips that character will chop off the last digit instead. (Use str() instead.)

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR:
- moves from rmutil to RLTest, and breaks the big test_modules.py into separate per command/feature test files. 
- It also moves from python2.7 to python3 ( fixes #595 ) -- cc @rafie 
- It enables AOF and replication tests ( and paves the road for OSS tests ).

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -166,25 +172,23 @@ $(TARGET): $(BIN_DIRS) $(OBJECTS) $(LIBRMUTIL)
 #----------------------------------------------------------------------------------------------
 
 TEST_REPORT_DIR ?= $(PWD)
-TEST_ARGS += \
-	--junitxml=$(TEST_REPORT_DIR)/tests_report.xml
+
+export GEN ?= 1
+export SLAVES ?= 1
+export AOF ?= 1
+export CLUSTER ?= 0
```
Comment (rafie) [inline src/Makefile:179]: From RediAI:
```
ifeq ($(SIMPLE),1)
export GEN=1
export SLAVES=0
export AOF=0
export CLUSTER=0
else
export GEN ?= 1
export SLAVES ?= 1
export AOF ?= 1
export CLUSTER ?= 1
endif
```

File: system-setup.py [hunk 2]
```diff
@@ -22,23 +22,40 @@ def common_first(self):
...
+            self.pip_install("psutil")
 
     def fedora(self):
         self.group_install("'Development Tools'")
+        self.install("python3 python3-psutil python3-networkx")
+
 
     def macos(self):
         if sh('xcode-select -p') == '':
             fatal("Xcode tools are not installed. Please run xcode-select --install.")
```
Comment (rafie) [inline system-setup.py:46]: We can remove this as we're now checking it inside Paella/Setup.

**Comment by rafie** [comment]
@filipecosta90 Great job! And it all works with Python 3!

**Comment by filipecosta90** [comment]
> @filipecosta90 Great job! And it all works with Python 3!

Thank you Rafi :) the move from 2 to 3 was really needed :)

**Review by rafie — DISMISSED** [review]

**Review by rafie — APPROVED** [review]

**Review by rafie — DISMISSED** [review]

File: system-setup.py [hunk 3]
```diff
@@ -22,23 +22,35 @@ def common_first(self):
...
     def common_last(self):
+        self.run("python3 -m pip uninstall -y RLTest || true")
         self.install("lcov")
         if not self.has_command("ramp"):
             self.pip_install("git+https://github.com/RedisLabs/RAMP@master")
-        self.pip_install("-r tests/requirements.txt")
+        if not self.has_command("RLTest"):  
+            # redis-py-cluster should be installed from git due to redis-py dependency
+            self.pip_install("--no-cache-dir git+https://github.com/Grokzen/redis-py-cluster.git@master")
+            self.pip_install("--no-cache-dir git+https://github.com/RedisLabsModules/RLTest.git@master")
```
Comment (danni-m) [inline system-setup.py:51]: its better to use a static version instead of master to make sure we don't break due to a new RLTest version

Comment (danni-m) [reply system-setup.py:51]: Also, im not sure this is needed as we use `install -r ` in line #53

File: src/unittests_parse_policies.c [hunk 4]
```diff
@@ -1,59 +1,58 @@
 /*
-* Copyright 2018-2020 Redis Labs Ltd. and Contributors
-*
-* This file is available under the Redis Labs Source Available License Agreement
-*/
-#include <stdio.h>
-#include <stdlib.h>
-#include "parse_policies.h"
-#include "minunit.h"
+ * Copyright 2018-2020 Redis Labs Ltd. and Contributors
```
Comment (danni-m) [inline src/unittests_parse_policies.c:2]: @filipecosta90 its best to have a different PR for whitespaces changes.

File: tests/flow/tests.sh [hunk 5]
```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply tests/flow/tests.sh:1]: yep. We're trying to standardize testing https://github.com/RedisAI/RedisAI/blob/master/tests/flow/tests.sh

File: tests/flow/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+pytest==3.0.7
+redis>=3.0.0
+git+https://github.com/Grokzen/redis-py-cluster.git@master
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply tests/flow/requirements.txt]: @danni-m updated redis-py-cluster. 
Regarding RLTest discussed even today with @rafie that given RLTest is pretty stable and fully controlled by ourselfs, for now we can update from master to keep this package always with the latest features.

File: src/unittests_parse_policies.c (hunk 4 above)
Comment (filipecosta90) [reply src/unittests_parse_policies.c:2]: @danni-m I've simply applied clang-format. Checking why we're not doing it constantly...

File: src/unittests_parse_duplicate_policy.c [hunk 7]
```diff
@@ -21,7 +21,31 @@ MU_TEST(test_duplicate_policy_to_string) {
     mu_check(strcmp(DuplicatePolicyToString(DP_MIN), "min") == 0);
 }
 
+MU_TEST(test_ParseCompactionPolicy) {
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply src/unittests_parse_duplicate_policy.c]: you're totally right @danni-m . moved the negative compaction policies tests to unittests_parse_policies.c

File: system-setup.py (hunk 3 above)
Comment (filipecosta90) [reply system-setup.py]: fixed as per your comment :+1:

File: src/unittests_parse_policies.c (hunk 4 above)
Comment (filipecosta90) [reply src/unittests_parse_policies.c:2]: @danni-m we had this wrong formating due to missing clang-format clauses for test files. Fixed this on commit so we don't have this behaviour in the future https://github.com/RedisTimeSeries/RedisTimeSeries/pull/605/commits/29715b9eda6068e7325709976ef26f49e37c9a9a

File: tests/flow/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+git+https://github.com/RedisLabsModules/RLTest.git@master
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply tests/flow/requirements.txt:4]: @danni-m updated redis-py-cluster.
Regarding RLTest discussed even today with @rafie that given RLTest is pretty stable and fully controlled by ourselfs, for now we can update from master to keep this package always with the latest features.

**Review by danni-m — APPROVED** [review]

**Pull request merged**